src/server/journalism.js

```diff
@@ -45,66 +90,113 @@ export const journalismTeam = (res, team) => {
+const extractClubInfo = (rawClubJson, team) => {
   const defaultObject = { abstract: 'No description available.' };
 
   try {
     const clubInfoRaw = rawClubJson.results.bindings[0];
     return {
       ...defaultObject,
-      abstract: clubInfoRaw.abstract.value,
+      abstract: clubInfoRaw.abstract,
     };
   } catch (err) {
     console.log(`Unable to retrieve club info for ${team}.`);
     return defaultObject;
   }
 };
 
-const extractGroundsInfo = (rawGroundsJson) => {
-  const defaultObject = { name: 'No name available.', thumbnail: 'none' };
+const extractGroundsInfo = (rawGroundsJson, team) => {
+  const defaultObject = { name: { value: 'No name available.' }, thumbnail: { value: 'none' } };
 
   try {
     const groundsInfoRaw = rawGroundsJson.results.bindings[0];
     return {
       ...defaultObject,
-      name: groundsInfoRaw.groundname.value,
-      thumbnail: groundsInfoRaw.thumbnail.value,
+      name: groundsInfoRaw.groundname,
+      thumbnail: groundsInfoRaw.thumbnail,
     };
   } catch (err) {
     console.log(`Unable to retrieve grounds info for ${team}.`);
     return defaultObject;
   }
 };
 
+export const journalismPlayer = (res, playerName) => {
+  const mainDescription = client
+    .query(`
+      SELECT DISTINCT * WHERE {
+        ?player foaf:name "${playerName}"@en .
+        {
+          ?player a umbel-rc:SoccerPlayer
+        } UNION {
+          ?player a dbo:SoccerPlayer
+        } .
+
+        OPTIONAL { ?player <http://dbpedia.org/property/fullname> ?fullname } .
+        OPTIONAL { ?player <http://dbpedia.org/ontology/Person/height> ?height } .
+        OPTIONAL { ?player <http://dbpedia.org/ontology/birthDate> ?birthdate } .
+        OPTIONAL { ?player <http://dbpedia.org/property/position> ?position . ?position rdfs:label ?positionlabel . FILTER(langMatches(lang(?positionlabel), "EN")) } .
+        OPTIONAL { ?player <http://dbpedia.org/property/caps> ?caps } .
+        OPTIONAL { ?player <http://dbpedia.org/property/quote> ?quote } .
+        OPTIONAL { ?player <http://dbpedia.org/ontology/thumbnail> ?thumbnail } .
+        OPTIONAL { ?player <http://dbpedia.org/property/currentclub> ?currentclub . ?currentclub <http://dbpedia.org/property/fullname> ?currentclubname } .
+
+        OPTIONAL { ?player dbo:abstract ?abstract . FILTER(langMatches(lang(?abstract), "EN")) } .
+      }
+      LIMIT 1
+    `)
+    .execute();
+
+  const allTeams = client
+    .query(`
+      SELECT DISTINCT * WHERE {
+        ?player foaf:name "${playerName}"@en .
+        {
+          ?player a umbel-rc:SoccerPlayer
+        } UNION {
+          ?player a dbo:SoccerPlayer
+        } .
+
+        ?player dbo:team ?team .
+        ?team rdfs:label ?teamname .
+        FILTER(langMatches(lang(?teamname), "EN")) .
+      }
+    `)
+    .execute();
+
+  Promise.all([mainDescription, allTeams])
+    .then((result) => {
+      try {
+        res.end(JSON.stringify({
+          description: result[0].results.bindings[0],
```

Comment:
what is this bindings business ? don't remember you having to do in journalism team